### ISSUE TITLE
Error: Missing host permission for the tab

### ISSUE THREAD

**Issue by idocaremucho** [az0/linkgopher#10]
Hi, this error is recently happening while extracting all the links from any page source, while extraction for the page itself is OK. ( not the source)
Anything I can do differently to make it work from the page source as before?
Thanks

**Comment by az0** [comment]
Yes, I see the issue. The web browser does not allow the extension to allow access  to view-source:* URLs, and at the moment I do not see how to fix this.

**Comment by idocaremucho** [comment]
Thanks for replying Andrew.

Have a good day.

Andrew Ziem wrote:

**Comment by FredThompsonII** [comment]
Is there a solution now? I'm using FF 56.0.2 and was using Link Gopher 1.3.3.1 but it's now giving an empty page so I installed build 2.0 which gives this access error.

**Comment by az0** [comment]
Earlier I reached out to some developers, but I did not get a response. I have not had time to track it down more, but I believe this a limitation of the new extension system for both Firefox and Google Chrome.

**Comment by FredThompsonII** [comment]
version 1.3.3.1 works properly with FF 56.x but version 2.0 does not.

links on "normal" web pages are displayed, self-hosted pages, such as those created by Youtube Checker https://addons.mozilla.org/en-US/firefox/addon/youtube-subscription-checker throw the error.

Grrr....

This means version 1.3.3.1 is the version to use because the desired results are created but the processing time for the creation of links extends relative to the number of links on a page.

**Comment by az0** [comment]
@FredThompsonII 

Yes, version 1.3.3.1 uses Firefox's old extension system that they no longer support, so it does not have the limitation, which seems like a security restriction.

**Comment by FredThompsonII** [comment]
...hate that about Mozilla. They WERE a web browser company...

What can I do to speed up 1.3.3.1 with FF 56.0.2 unbranded?

**Comment by az0** [comment]
@FredThompsonII  I am not sure what you mean "speed up." Do you mean it takes a long time to load the links? In one of your comments, you wrote that it gives you a blank page, and then in another comment you wrote that it works

I can no longer support Link Gopher version 1.x, and starting with Firefox 57, Mozilla no longer supporters the "legacy" extensions system that Link Gopher version 1.x used.

I already tried reaching out for some help with other developers but did not get any response. If anyone is interested in funding a solution, I would manage hiring a contractor. This assumes that something can be done in the extension itself and is not an inherit limitation of the new WebExtensions system that Mozilla and Google provide.

**Comment by P4z** [comment]
I have got this missing permission error when trying to _Extract All Links_ not from the `view-source:*` address but from https://addons.mozilla.org/ page. It works fine on GitHub, YouTube and other so long.
